include/deal.II/lac/solver_cg.h

```diff
@@ -166,6 +182,33 @@ class SolverCG : public Solver<VECTOR>
          const VECTOR         &b,
          const PRECONDITIONER &precondition);
 
+  /**
+   * Connect a slot to retrieve the CG coefficients. The slot will be called
+   * with alpha as the first argument and with beta as the second argument.
+   * Called once per iteration.
+   */
+  boost::signals2::connection
+  connect_coefficients_slot(
+    const std_cxx11::function<void (double,double)> &slot);
+
+  /**
+   * Connect a slot to retrieve the estimated condition number.
+   * Called on each iteration if every_iteration=true otherwise called
+   * after all iterations.
+   */
+  boost::signals2::connection
+  connect_condition_number_slot(const std_cxx11::function<void (double)> &slot,
+                                const bool every_iteration=false);
+
+  /**
+   * Connect a slot to retrieve the estimated eigenvalues.
+   * Called on each iteration if every_iteration=true otherwise called
+   * after all iterations.
+   */
+  boost::signals2::connection
+  connect_eigenvalues_slot(const std_cxx11::function<EigenvalueSlot> &slot,
+                           const bool every_iteration=false);
```

Comment:
Since you spell out the type of the function everywhere else, I would avoid the typedef `Eigenvalue` slot here as well. This also makes it clear immediately (without having to look up the function signature) what the function arguments are.

Same with "after all iterations" as above.